uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/instances/leave-request.instance.js

```diff
@@ -11,6 +11,71 @@ define([
   instances.factory('LeaveRequestInstance', ['$q', 'checkPermissions', 'OptionGroup',
     'shared-settings', 'ModelInstance', 'LeaveRequestAPI',
     function ($q, checkPermissions, OptionGroup, sharedSettings, ModelInstance, LeaveRequestAPI) {
+      /**
+       * Amends balance change breakdown if calculation unit absence type is "hours".
+       * It changes the balance first and last days (only first day in case of single day request)
+       *   by setting the selected deductions.
+       * @NOTE this function mutates the "balanceChange" object
+       *
+       * @param  {Object} balanceChange
+       * @param  {String} calculationUnit
+       * @return {Promise} resolve with amended balance change in case of "hours" absence type
+       *   or returns unamended balance change in case of "days" absence type
+       */
+      function amendHourlyBalanceChangeBreakdown (balanceChange, calculationUnit) {
+        var breakdown = balanceChange.breakdown;
+        var request = this;
+
+        if (calculationUnit !== 'hours') {
+          return balanceChange;
```

Comment:
The doc says the function returns promise, is it a promise?

Reply:
Maybe you can wrap this in a `$q.resolve()`

Reply:
Or do you even need to return anything? You can just call `return`.

Reply:
I will wrap in promise, it needs to return balance.

Reply:
Upd: I dont need `return balanceChange` anymore. So simply removed it.

Reply:
done.